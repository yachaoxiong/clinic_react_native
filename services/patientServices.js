const { getToken } = require('../utils/functions');

export const getAllPatients = async () => {
  const token = await getToken();
  console.log("token->", token);
  return fetch('https://yachao-clinic-app.herokuapp.com/api/patients', {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer ' + token
    }
  })
    .then((response) => response.json())
    .then((responseJson) => {
      return responseJson;
    })
    .catch((error) => {
      console.error(error);
    }
  );
}