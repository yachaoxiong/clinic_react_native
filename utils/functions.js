import * as SecureStore from 'expo-secure-store';
//calculate BMI BY WEIGHT(KG) AND HEIGHT(CM)
export const calculateBMI = (weight, height) => {

    // calculate BMI
    const bmi =  weight / (height / 100) ** 2 

  
   // convert to one decimal place
   const convertedBmi = bmi.toFixed(1);

  // set result message
  const result = convertedBmi < 18.5 
   ? 'Underweight'
   : (convertedBmi > 18.5 && convertedBmi <= 24.9) 
   ? 'Normal' 
   : (convertedBmi >= 25 && convertedBmi <= 29.9) 
   ? 'Overweight'
  : 'Obese';
  
  return { bmi: convertedBmi, result };
}

export const setToken = async (token) => {
  try {
    await SecureStore.setItemAsync('token', token);
  } catch (error) {
    console.log('Error setting token', error);
  }
}

export const getToken = async () => {
  try {
    const token = await SecureStore.getItemAsync('token');
    return token;
  } catch (error) {
    console.log('Error getting token', error);
  }
}

export const removeToken = async () => {
  try {
    await SecureStore.deleteItemAsync('token');
  } catch (error) {
    console.log('Error removing token', error);
  }
}


