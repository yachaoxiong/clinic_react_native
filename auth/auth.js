import { setToken, getToken } from '../utils/functions';
import fetch from 'node-fetch';

export const login = ( username, password) => {
  return fetch('https://yachao-clinic-node.herokuapp.com/api/auth/login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      username: username,
      password: password
    })
  })
    .then((response) => response.json())
    .then(async (responseJson) => {
      if (responseJson.token) {
        await setToken(responseJson.token);
        return responseJson;
      }
      return "error";
      
    })
    .catch((error) => {
      console.error(error);
      return "error";
    }
  );
}

export const register = (username, password, email, role) => {
  return fetch('https://yachao-clinic-node.herokuapp.com/api/auth/register', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      username: username,
      password: password,
      email: email,
      role: role
    })
  })
    .then((response) => response.json())
    .then((responseJson) => {
      if (responseJson.message) {
        return "error";
      }
      return "success";
    })
    .catch((error) => {
      console.error(error);
      return "error";
    }
  );
}


export const getUser = async () => {
  const token = await getToken();
  try {
    const response = await fetch('https://yachao-clinic-node.herokuapp.com/api/auth/getUser', {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + token
      }
    })
    const responseJson = await response.json()
    return responseJson;
  } catch (error) {
    console.log(error);
    return null;
  }
}
