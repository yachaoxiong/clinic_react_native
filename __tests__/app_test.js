
import { login, register } from "../auth/auth";
import { getAllPatients, updatePatient, addNewPatient,deletePatient, getPatientHistory, getPatientById, addNewRecord,getAllTestsByPatientId} from "../services/patientServices";
import { setToken, removeToken, getToken, calculateBMI } from "../utils/functions";

const mockPatient = {
  "firstName" : "Lily",
  "lastName" : "Mee",
  "age":28,
  "birthDate": "1994-08-26",
  "address" : "228 t crescent",
  "weight" : 60,
  "height" : 176,
  "gender": "female",
  "email":"Lily2@gmail.com",
  "phone":"223-666-9999",
  "department" : "emergency",
  "status":"normal",
  "doctor" : "635ed207b9036eccbe9d665b"
}

const mockTest = {
  "category":"Blood Pressure",
  "readings":[
      {
          "name":"diatomic",
          "value":60
      },
      {
          "name":"systolic",
          "value":190
      }
  ],
  "isCritical":true
}


// mock functions
jest.mock('../utils/functions', () => {
  return {
    getToken: jest.fn(),
    setToken: jest.fn(),
    removeToken: jest.fn(),
    calculateBMI: jest.fn(),
  };
});


const testUser = {
    "username":"William12",
    "password" : "as123",
    "email":"William2@gmail.com",
    "role":"doctor"
}
let userToken = null;
jest.mock('expo-secure-store', () => {
  require('../utils/functions');
  return {
    getItemAsync: jest.fn(),
    setItemAsync: jest.fn(),
    deleteItemAsync: jest.fn(),
  };
});

describe("register", () => {
  it("should return error if username is taken", async () => {
    const response = await register(testUser.username, testUser.password, testUser.email, testUser.role);
    console.log("response", response)
    expect(response).toBe("error");

  });
});


describe("login", () => {
  it("should return a token and user profile", async () => {
    const response = await login(testUser.username, testUser.password);
    expect(response.token).not.toBeNull();
    userToken = response.token;
  });
  it("should return error", async () => {
    const response = await login(testUser.username, "wrong password");
    expect(response).toBe("error");
  });
});


//utils functions
describe("utils Functions", () => {
  it("should set token", async () => {
    // USE THIS TO MOCK THE FUNCTION
    setToken.mockImplementation(() => {
      return "success";
    });
    
    const response = await setToken(userToken);
    expect(response).toBe("success");

  });

  it("should get token", async () => {
    getToken.mockImplementation(() => {
      return userToken;
    });
    const response = await getToken();
    expect(response).toBe(userToken);
  });

  it("should remove token", async () => {
    removeToken.mockImplementation(() => {
      return "success";
    });
    const response = await removeToken();
    expect(response).toBe("success");
  });


  it("should calculate BMI", () => {
    calculateBMI.mockImplementation((height, weight) => {
          // calculate BMI
      const bmi =  weight / (height / 100) ** 2 

    
      // convert to one decimal place
      const convertedBmi = bmi.toFixed(1);
  
    // set result message
    const result = convertedBmi < 18.5 
      ? 'Underweight'
      : (convertedBmi > 18.5 && convertedBmi <= 24.9) 
      ? 'Normal' 
      : (convertedBmi >= 25 && convertedBmi <= 29.9) 
      ? 'Overweight'
    : 'Obese';
    
        return { bmi: convertedBmi, result };
    })
    const result = calculateBMI(180, 70);
    expect(result.bmi).toBe("21.6");
    expect(result.result).toBe("Normal");
  });
  
})


//services
describe("services", () => {
 
  it("should get all patients", async () => {
    const response = await getAllPatients();
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be array
    expect(Array.isArray(response)).toBe(true);
    
  });
  it("should Create a new patient", async () => {
    const response = await addNewPatient(mockPatient)
   
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be object
    expect(typeof response).toBe("object");

    expect(response.firstName).toBe(mockPatient.firstName);
    
    // has new _id
    expect(response._id).not.toBeUndefined();

    mockPatient._id = response._id;
  });

  it("should update a patient", async () => {
    mockPatient.firstName = "Lily2";
    const response = await updatePatient(mockPatient)
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be object
    expect(typeof response).toBe("object");

    expect(response.firstName).toBe(mockPatient.firstName);
  });
  it("should add a test to a patient and automatically update patient status if it is critical", async () => {
    const response = await addNewRecord(mockPatient._id, mockTest)
    const patient = await getPatientById(mockPatient._id)
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be object
    expect(typeof response).toBe("object");
    //should have a id
    expect(response._id).not.toBeUndefined();

    expect(patient.status).toBe("critical");

  })
  it("should get a patient by id", async () => {
    const response = await getPatientById(mockPatient._id)
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be object
    expect(typeof response).toBe("object");
    //should have a id
    expect(response._id).not.toBeUndefined();
  })

  it("should get all tests for a patient", async () => {
    const response = await getAllTestsByPatientId(mockPatient._id)
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be array
    expect(Array.isArray(response)).toBe(true);
  })

  it("should get the history of a patient", async () => {
    const response = await getPatientHistory(mockPatient._id)
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be array
    expect(Array.isArray(response)).toBe(true);
  })
  
  it("should delete a patient", async () => {
    const response = await deletePatient(mockPatient._id)
    //should have a message
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    //not undefined
    expect(response).not.toBeUndefined();
    //should be object
    expect(typeof response).toBe("object");
    //should have a message
    expect(response.message).toEqual("Patient successfully deleted!");
  })
});





