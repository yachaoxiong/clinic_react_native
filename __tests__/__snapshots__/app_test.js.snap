// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "flex": 1,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "backgroundColor": "rgb(242, 242, 242)",
                                  "flex": 1,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                {
                                  "zIndex": 1,
                                }
                              }
                            >
                              <View
                                accessibilityElementsHidden={false}
                                importantForAccessibility="auto"
                                onLayout={[Function]}
                                pointerEvents="box-none"
                                style={null}
                              >
                                <View
                                  collapsable={false}
                                  pointerEvents="box-none"
                                  style={
                                    {
                                      "bottom": 0,
                                      "left": 0,
                                      "opacity": 1,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                      "zIndex": 0,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    style={
                                      {
                                        "backgroundColor": "rgb(255, 255, 255)",
                                        "borderBottomColor": "rgb(216, 216, 216)",
                                        "borderBottomWidth": 0,
                                        "elevation": 0,
                                        "flex": 1,
                                        "shadowColor": "rgb(216, 216, 216)",
                                        "shadowOffset": {
                                          "height": 0.5,
                                          "width": 0,
                                        },
                                        "shadowOpacity": 0,
                                        "shadowRadius": 0,
                                      }
                                    }
                                  />
                                </View>
                                <View
                                  collapsable={false}
                                  pointerEvents="box-none"
                                  style={
                                    {
                                      "height": 44,
                                      "maxHeight": undefined,
                                      "minHeight": undefined,
                                      "opacity": undefined,
                                      "transform": undefined,
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="none"
                                    style={
                                      {
                                        "height": 0,
                                      }
                                    }
                                  />
                                  <View
                                    pointerEvents="box-none"
                                    style={
                                      {
                                        "alignItems": "stretch",
                                        "flex": 1,
                                        "flexDirection": "row",
                                      }
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                      pointerEvents="box-none"
                                      style={
                                        {
                                          "alignItems": "flex-start",
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "justifyContent": "center",
                                          "marginStart": 0,
                                          "opacity": 1,
                                        }
                                      }
                                    />
                                    <View
                                      collapsable={false}
                                      pointerEvents="box-none"
                                      style={
                                        {
                                          "justifyContent": "center",
                                          "marginHorizontal": 16,
                                          "maxWidth": 718,
                                          "opacity": 1,
                                        }
                                      }
                                    >
                                      <Text
                                        accessibilityRole="header"
                                        aria-level="1"
                                        collapsable={false}
                                        numberOfLines={1}
                                        onLayout={[Function]}
                                        style={
                                          {
                                            "color": "rgb(28, 28, 30)",
                                            "fontSize": 17,
                                            "fontWeight": "600",
                                          }
                                        }
                                      >
                                          
                                      </Text>
                                    </View>
                                    <View
                                      collapsable={false}
                                      pointerEvents="box-none"
                                      style={
                                        {
                                          "alignItems": "flex-end",
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "justifyContent": "center",
                                          "marginEnd": 0,
                                          "opacity": 1,
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={
                                {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                pointerEvents="box-none"
                                style={
                                  {
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  }
                                }
                              >
                                <View
                                  collapsable={false}
                                  style={
                                    {
                                      "opacity": 1,
                                    }
                                  }
                                />
                                <View
                                  accessibilityElementsHidden={false}
                                  closing={false}
                                  gestureVelocityImpact={0.3}
                                  importantForAccessibility="auto"
                                  onClose={[Function]}
                                  onGestureBegin={[Function]}
                                  onGestureCanceled={[Function]}
                                  onGestureEnd={[Function]}
                                  onOpen={[Function]}
                                  onTransition={[Function]}
                                  pointerEvents="box-none"
                                  style={
                                    [
                                      {
                                        "display": "flex",
                                        "overflow": undefined,
                                      },
                                      {
                                        "bottom": 0,
                                        "left": 0,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 0,
                                      },
                                    ]
                                  }
                                  transitionSpec={
                                    {
                                      "close": {
                                        "animation": "spring",
                                        "config": {
                                          "damping": 500,
                                          "mass": 3,
                                          "overshootClamping": true,
                                          "restDisplacementThreshold": 10,
                                          "restSpeedThreshold": 10,
                                          "stiffness": 1000,
                                        },
                                      },
                                      "open": {
                                        "animation": "spring",
                                        "config": {
                                          "damping": 500,
                                          "mass": 3,
                                          "overshootClamping": true,
                                          "restDisplacementThreshold": 10,
                                          "restSpeedThreshold": 10,
                                          "stiffness": 1000,
                                        },
                                      },
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    pointerEvents="box-none"
                                    style={
                                      {
                                        "flex": 1,
                                      }
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                      forwardedRef={[Function]}
                                      handlerTag={2}
                                      handlerType="PanGestureHandler"
                                      needsOffscreenAlphaCompositing={false}
                                      onGestureHandlerEvent={[Function]}
                                      onGestureHandlerStateChange={[Function]}
                                      style={
                                        {
                                          "flex": 1,
                                          "transform": [
                                            {
                                              "translateX": 0,
                                            },
                                            {
                                              "translateX": 0,
                                            },
                                          ],
                                        }
                                      }
                                    >
                                      <View
                                        pointerEvents="box-none"
                                        style={
                                          [
                                            {
                                              "flex": 1,
                                              "overflow": "hidden",
                                            },
                                            [
                                              {
                                                "backgroundColor": "rgb(242, 242, 242)",
                                              },
                                              undefined,
                                            ],
                                          ]
                                        }
                                      >
                                        <View
                                          style={
                                            {
                                              "flex": 1,
                                              "flexDirection": "column-reverse",
                                            }
                                          }
                                        >
                                          <View
                                            style={
                                              {
                                                "flex": 1,
                                              }
                                            }
                                          >
                                            <View
                                              style={
                                                {
                                                  "alignItems": "center",
                                                  "backgroundColor": "#fff",
                                                  "flex": 1,
                                                  "paddingHorizontal": 30,
                                                }
                                              }
                                            >
                                              <Text
                                                style={
                                                  {
                                                    "color": "#383838",
                                                    "fontSize": 28,
                                                    "fontWeight": "bold",
                                                  }
                                                }
                                              >
                                                Login
                                              </Text>
                                              <Image
                                                source={1}
                                                style={
                                                  {
                                                    "height": 160,
                                                    "resizeMode": "contain",
                                                    "width": "100%",
                                                  }
                                                }
                                              />
                                              <View
                                                style={
                                                  {
                                                    "alignItems": "center",
                                                    "flexDirection": "row",
                                                    "justifyContent": "space-between",
                                                    "paddingVertical": 10,
                                                  }
                                                }
                                              >
                                                <TextInput
                                                  onBlur={[Function]}
                                                  onChangeText={[Function]}
                                                  onFocus={[Function]}
                                                  placeholder="username"
                                                  style={
                                                    {
                                                      "backgroundColor": "#f5f6fa",
                                                      "borderColor": "transparent",
                                                      "borderRadius": 5,
                                                      "borderWidth": 2,
                                                      "color": "#383838",
                                                      "fontSize": 18,
                                                      "padding": 11,
                                                      "width": "100%",
                                                    }
                                                  }
                                                  value=""
                                                />
                                              </View>
                                              <View
                                                style={
                                                  {
                                                    "alignItems": "center",
                                                    "flexDirection": "row",
                                                    "justifyContent": "space-between",
                                                    "paddingVertical": 10,
                                                  }
                                                }
                                              >
                                                <TextInput
                                                  onBlur={[Function]}
                                                  onChangeText={[Function]}
                                                  onFocus={[Function]}
                                                  placeholder="password"
                                                  secureTextEntry={true}
                                                  style={
                                                    {
                                                      "backgroundColor": "#f5f6fa",
                                                      "borderColor": "transparent",
                                                      "borderRadius": 5,
                                                      "borderWidth": 2,
                                                      "color": "#383838",
                                                      "fontSize": 18,
                                                      "padding": 11,
                                                      "width": "100%",
                                                    }
                                                  }
                                                  value=""
                                                />
                                              </View>
                                              <Text
                                                style={
                                                  {
                                                    "color": "#e74c3c",
                                                    "fontSize": 14,
                                                    "fontWeight": "bold",
                                                    "paddingVertical": 10,
                                                  }
                                                }
                                              />
                                              <View
                                                accessible={true}
                                                collapsable={false}
                                                focusable={true}
                                                onClick={[Function]}
                                                onResponderGrant={[Function]}
                                                onResponderMove={[Function]}
                                                onResponderRelease={[Function]}
                                                onResponderTerminate={[Function]}
                                                onResponderTerminationRequest={[Function]}
                                                onStartShouldSetResponder={[Function]}
                                                style={
                                                  {
                                                    "alignItems": "center",
                                                    "backgroundColor": "#052F5F",
                                                    "borderRadius": 2,
                                                    "justifyContent": "center",
                                                    "opacity": 1,
                                                    "paddingHorizontal": 10,
                                                    "paddingVertical": 15,
                                                    "width": 180,
                                                  }
                                                }
                                              >
                                                <Text
                                                  style={
                                                    {
                                                      "color": "#fff",
                                                      "fontSize": 16,
                                                      "fontWeight": "bold",
                                                    }
                                                  }
                                                >
                                                  Login
                                                </Text>
                                              </View>
                                              <View
                                                style={
                                                  {
                                                    "alignItems": "center",
                                                    "flexDirection": "row",
                                                    "justifyContent": "center",
                                                    "marginTop": 20,
                                                  }
                                                }
                                              >
                                                <Text
                                                  style={
                                                    {
                                                      "color": "#383838",
                                                      "fontSize": 16,
                                                    }
                                                  }
                                                >
                                                  Don't have an account?
                                                </Text>
                                                <View
                                                  accessible={true}
                                                  collapsable={false}
                                                  focusable={true}
                                                  onClick={[Function]}
                                                  onResponderGrant={[Function]}
                                                  onResponderMove={[Function]}
                                                  onResponderRelease={[Function]}
                                                  onResponderTerminate={[Function]}
                                                  onResponderTerminationRequest={[Function]}
                                                  onStartShouldSetResponder={[Function]}
                                                  style={
                                                    {
                                                      "opacity": 1,
                                                    }
                                                  }
                                                >
                                                  <Text
                                                    style={
                                                      {
                                                        "color": "#052F5F",
                                                        "fontSize": 16,
                                                        "marginLeft": 5,
                                                      }
                                                    }
                                                  >
                                                     Register
                                                  </Text>
                                                </View>
                                              </View>
                                            </View>
                                          </View>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
